test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemTest.scala

```diff
@@ -0,0 +1,155 @@
+
+  @Test def setOut(): Unit = {
+    val savedOut = System.out
+    try {
+      val testOut = new java.io.PrintStream(new java.io.ByteArrayOutputStream)
+      System.setOut(testOut)
+      assertTrue(System.out eq testOut)
+    } finally {
+      System.setOut(savedOut)
+    }
+  }
+
+  @Test def setErr(): Unit = {
+    val savedErr = System.err
+    try {
+      val testErr = new java.io.PrintStream(new java.io.ByteArrayOutputStream)
+      System.setErr(testErr)
+      assertTrue(System.err eq testErr)
+    } finally {
+      System.setErr(savedErr)
+    }
+  }
+
+  @Test def should_respond_to_arraycopy(): Unit = {
+    val object0 = Array[Any]("[", "b", "c", "d", "e", "f", "]")
+    val object1 = Array[Any](() => true, 1, "2", '3', 4.0, true, object0)
+
+    System.arraycopy(object1, 1, object0, 1, 5)
+    if (executingInJVM) {
+      assertEquals("[1234.0true]", object0.mkString)
+    } else {
+      assertEquals("[1234true]", object0.mkString)
+    }
+
+    val string0 = Array("a", "b", "c", "d", "e", "f")
+    val string1 = Array("1", "2", "3", "4")
+
+    System.arraycopy(string1, 0, string0, 3, 3)
+    assertEquals("abc123", string0.mkString)
+
+    val ab01Chars = Array("ab".toCharArray, "01".toCharArray)
+    val chars = new Array[Array[Char]](32)
+    System.arraycopy(ab01Chars, 0, chars, 0, 2)
+    for (i <- Seq(0, 2, 4, 8, 16)) {
+      System.arraycopy(chars, i / 4, chars, i, i)
+    }
+
+    assertEquals(12, chars.filter(_ == null).length)
+    assertEquals("ab01ab0101ab01ab0101ab0101ab01ab0101ab01",
+      chars.filter(_ != null).map(_.mkString).mkString)
+
+  }
+
+  @Test def should_respond_to_arraycopy_with_range_overlaps_for_the_same_array(): Unit = {
+    val array = new Array[Int](10)
+
+    for (i <- 1 to 6) {
+      array(i) = i
+    }
+
+    assertArrayEquals(Array(0, 1, 2, 3, 4, 5, 6, 0, 0, 0), array)
+    System.arraycopy(array, 0, array, 3, 7)
+    assertArrayEquals(Array(0, 1, 2, 0, 1, 2, 3, 4, 5, 6), array)
+
+    System.arraycopy(array, 0, array, 1, 0)
+    assertArrayEquals(Array(0, 1, 2, 0, 1, 2, 3, 4, 5, 6), array)
+
+    System.arraycopy(array, 0, array, 1, 9)
+    assertArrayEquals(Array(0, 0, 1, 2, 0, 1, 2, 3, 4, 5), array)
+
+    System.arraycopy(array, 1, array, 0, 9)
+    assertArrayEquals(Array(0, 1, 2, 0, 1, 2, 3, 4, 5, 5), array)
+
+    System.arraycopy(array, 0, array, 0, 10)
+    assertArrayEquals(Array(0, 1, 2, 0, 1, 2, 3, 4, 5, 5), array)
+
+    val reversed = array.reverse
+    System.arraycopy(reversed, 5, array, 5, 5)
+    assertArrayEquals(Array(0, 1, 2, 0, 1, 1, 0, 2, 1, 0), array)
+  }
+
+  @Test def should_provide_identityHashCode(): Unit = {
+    /* This test is more restrictive than the spec, but we know our
+     * implementation will always pass the test.
+     */
+    class HasIDHashCode
+
+    val x1 = new HasIDHashCode
+    val x2 = new HasIDHashCode
+    val x1FirstHash = x1.hashCode()
+    assertEquals(x1FirstHash, x1.hashCode())
+    assertNotEquals(x1.hashCode(), x2.hashCode())
```

Comment:
So this line *could* fail on the JVM on unlucky runs. It should be protected by `!executingInJVM`